ddl/ddl_worker.go

```diff
@@ -348,7 +374,8 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx, shouldCleanJobs bool) error {
 			return errors.Trace(w.handleUpdateJobError(t, job, err))
 		})
 
-		if runJobErr != nil {
+		waitDependencyJob := job != nil && job.DependencyID != 0
+		if runJobErr != nil || waitDependencyJob {
```

Comment:
If waitDependencyJob is true, it is not an error.

Reply:
Yes, but we'd better wait a moment.